### ISSUE TITLE
Uncaught TypeError: Cannot read property 'match' of undefined

### ISSUE THREAD

**Issue by rhysburnie** [vt5491/multi-theme-applicator#6]
**Atom**: 1.19.6 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.12.6
**Thrown From**: [multi-theme-applicator](https://github.com/vt5491/multi-theme-applicator) package 1.2.0

### Stack Trace

Uncaught TypeError: Cannot read property 'match' of undefined

```
At /Users/rgb/.atom/packages/multi-theme-applicator/lib/utils.coffee:109

TypeError: Cannot read property 'match' of undefined
    at Utils.module.exports.Utils.getFileExt (/packages/multi-theme-applicator/lib/utils.coffee:109:26)
    at LocalThemeSelectorView.module.exports.LocalThemeSelectorView.applyLocalTheme (/packages/multi-theme-applicator/lib/local-theme-selector-view.coffee:261:24)
    at /packages/multi-theme-applicator/lib/local-theme-selector-view.coffee:118:21)
    at HTMLFormElement.dispatch (/packages/multi-theme-applicator/node_modules/jquery/dist/jquery.js:4737:27)
    at HTMLFormElement.elemData.handle (/packages/multi-theme-applicator/node_modules/jquery/dist/jquery.js:4549:28)
```

### Commands

```
     -0:21.5.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-when-scrolling)
     -0:17.2.0 multi-theme-applicator:toggle (div.panels-item)
```

### Non-Core Packages

```
atom-ternjs 0.18.3 
autocomplete-modules 1.6.10 
busy-signal 1.4.3 
docblockr 0.10.3 
editorconfig 2.2.2 
emmet 2.4.3 
es6-javascript 1.0.0 
file-icons 2.1.8 
intentions 1.1.2 
javascript-snippets 1.2.1 
language-babel 2.66.1 
language-vue 0.23.1 
linter 2.2.0 
linter-eslint 8.2.1 
linter-ui-default 1.6.1 
multi-theme-applicator 1.2.0 
prettier-atom 0.38.0 
project-manager 3.3.5 
sassdoc 0.1.0 
```

**Comment by vt5491** [comment]
Ok, thanks for letting me know.  I will investigate.  Does the problem occur consistently?  Is it blocking you from using MTA?  Does it happen when you try to apply a local theme? A brief look on my log, does not reveal any problems, but I'm running windows.  I have a mac box I can test on, as this might be a mac-only problem.

**Comment by rhysburnie** [comment]
Yeah as soon as I installed. Tried variations of context and with no context same result.

**Comment by vt5491** [comment]
Are you tying to invoke MTA on atom session that does not have any open files ie. like when the editor first comes up?  The message is saying that it can't find any filenames.  I am able to reproduce this problem if I invoke MTA on an atom session where I don't have any open files.  I get:

```
[Uncaught TypeError: Cannot read property 'match' of undefined
    at Utils.module.exports.Utils.getFileExt (utils.coffee:109)
    at LocalThemeSelectorView.module.exports.LocalThemeSelectorView.applyLocalTheme (local-theme-selector-view.coffee:261)
    at HTMLDivElement.LocalThemeSelectorView.subscriptions.add.atom.commands.add.local-theme-selector-view:applyLocalTheme (local-theme-selector-view.coffee:102)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (command-registry.coffee:256)
    at command-registry.coffee:1
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (C:\Users\vturner2\AppData\Local\atom\app-1.19.5\resources\app\node_modules\atom-keymap\lib\keymap-m…:606)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (C:\Users\vturner2\AppData\Local\atom\app-1.19.5\resources\app\node_modules\atom-keymap\lib\keymap-m…:397)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (window-event-handler.coffee:75)
    at HTMLDocument.<anonymous> (window-event-handler.coffee:1)](url)
```
Which is almost identical to the problem you report.

As currently written, MTA expects there to be at least one file open -- it's a syntax themer not a UI themer after all.  If this is what you're doing, try opening a file first and then invoke cmd-shift-v.

If this is the case, I will look into adding some kind of error message in this circumstance.  I suppose if someone were trying to apply a window level theme, then this might not be an obvious requirement.  In my day to day use I personally never use the window or pane level theming, and exclusively use either "file type" or "file".

I tested under Mac, and everything worked fine, so I don't think this has anything to do with the Mac.

**Comment by vt5491** [comment]
Also, make sure you're not trying to apply against a non-editor window, like the customization panel.

**Comment by rhysburnie** [comment]
yeah ok that was it

**Comment by vt5491** [comment]
Great.  I'll see what I can do to close down that use case.  Since I'm looking at the code, it's looking like I'm going to be making a few other fixes that I've discovered over time, so you might see a new release coming down the pike.

Thanks for the feedback and your interest in using MTA.  Please let me know if you have any other questions or issues.

**Comment by vt5491** [comment]
fixed in v 1.3.0
